11-trt-llm/README.md

```diff
@@ -0,0 +1,75 @@
+## TRT-LLM Builder Truss
+
+### Introduction
+TRT-LLM is high-performance language model inference runtime from Nvidia. To use TRT-LLM on a model, users have to _compile_ a TRT engine from their corresponding PyTorch weights. In this Truss, we provide a config-driven interface to build the engine. We build the engine as part of the server startup and then use Triton Inference Server to serve the engine for high-performance workloads.
+
+NOTE: Building an engine can take a very long time. The approach outlined above is not scalable for fast cold-starts or production workloads where autoscaling may cause multiple replicas to build the engine repeatedly.
+
+### TRT-LLM and Truss
+To use `TRT-LLM` in Truss, we provide a key `trt_llm` in the `config.yaml` users can use.
+```
+trt_llm:
+  build:
+    base_model_architecture: LLAMA  # Model architecture, e.g. LLAMA, MISTRAL, DEEPSEEK
+    max_input_len: 2048  # Maximum input sequence length
+    max_output_len: 256  # Maximum output sequence length 
+    max_batch_size: 8  # Maximum batch size for inference
+    max_beam_width: 4  # Maximum beam width for beam search
+    huggingface_ckpt_repository: "decaplusplus/llama-7b-hf" # Path to HuggingFace model checkpoint
+    quantization_type: WEIGHTS_KV_INT8 # Quantization type for the engine
+    tensor_parallel_count: 1 # Tensor parallelism count
+    pipeline_parallel_count: 1 # Pipeline parallelism count
+    plugin_configuration:
+      multi_block_mode: False  # Enable multi-block mode
+      paged_kv_cache: True  # Enable paged KV cache
+      use_fused_mlp: False  # Use fused MLP
+    
+  serve:
+    engine_repository: /path/to/engine  # Huggingface repository with TRT engine
+    tokenizer_repository: /path/to/tokenizer  # Huggingface repository with TRT engine
+    tensor_parallel_count: 1
+    pipeline_parallel_count: 1
+```
+
+The `build` section specifies the configuration for building the TRT engine from a PyTorch model checkpoint. The required fields are:
+
+- `base_model_architecture`: The model architecture (LLAMA, MISTRAL, DEEPSEEK)
+- `max_input_len`: Maximum input sequence length
+- `max_output_len`: Maximum output sequence length
+- `max_batch_size`: Maximum batch size for inference
+- `max_beam_width`: Maximum beam width for beam search
+- `huggingface_ckpt_repository`: Path to the HuggingFace model checkpoint with the PyTorch weights
+
+Other optional fields control quantization, parallelism, and plugin settings.
+
+The `serve` section specifies the paths to the compiled TRT engine and tokenizer to use for serving the model. 
+
+Note that build and serve are mutually exclusive - you can only specify one or the other in the config.
```

Comment:
Would be good to set this up at the top of the section: why would you write a build config and why would you write a serve config?